src/main/java/org/popcraft/popcraft/commands/Me.java

```diff
@@ -1,42 +1,39 @@
 package org.popcraft.popcraft.commands;
 
-import org.bukkit.Bukkit;
+import com.google.common.base.Joiner;
+import com.google.inject.Inject;
+import org.bukkit.Server;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
 import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.entity.Player;
+import org.popcraft.popcraft.newCode.PopCommand;
 import org.popcraft.popcraft.utils.Message;
 
 import java.io.File;
 
-@Deprecated
+@PopCommand("me")
 public class Me implements CommandExecutor {
 
+    private final Server server;
+
+    @Inject
+    public Me(Server server) {
+        this.server = server;
+    }
+
     @Override
     public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
-        Player player = (Player) sender;
-        if (cmd.getName().equalsIgnoreCase("me")) {
-            String player1 = sender.getName();
-            File file = new File("plugins/Essentials/userdata/" + player.getUniqueId() + ".yml");
-            YamlConfiguration userdata = YamlConfiguration.loadConfiguration(file);
-            if (!userdata.contains("muted") || (userdata.contains("muted") && !userdata.getBoolean("muted"))) {
-                String message = "";
-                int a = 0;
-                if (args.length > 0) {
-                    while (a < args.length) {
-                        message = (message + args[a] + " ");
-                        a++;
-                    }
-                    Bukkit.broadcastMessage("* " + player1 + " " + message);
-                } else {
-                    Message.usage(player, "me <action>");
-                }
-            } else {
-                Message.normal(player, "Your voice has been silenced!");
-            }
-            return true;
-        }
-        return false;
+        if (!(sender instanceof Player) || (args.length < 1))
+            return false;
+        final Player player = (Player) sender;
+        File playerFile = new File("plugins/Essentials/userdata/" + player.getUniqueId() + ".yml");

```

Comment:
This is fine for now but just as note, maybe we can look into the EssentialX API to see if there's a better way to implement this